array/DNAm/analysis/EWAS/summariseWithinCTDMPs.r

```diff
@@ -0,0 +1,272 @@
+
+## COMPARE DMPS ACROSS CELL TYPES
+thres<-1e-5
+
+dmpList<-NULL
+for(i in 1:3){
+    dmpList<-rbind(dmpList, cbind(rownames(res[[i]])[which(res[[i]][,"FullModel_SCZ_P"] < thres)], cellTypes[i]))
+}
+colnames(dmpList)<-c("ProbeID", "DiscoveryCellType")
+
+
+
+dmpRes<-cbind(dmpList, res[[1]][dmpList[,1], c("FullModel_SCZ_P", "FullModel_SCZ_coeff", "FullModel_SCZ_SE")],
+res[[2]][dmpList[,1], c("FullModel_SCZ_P", "FullModel_SCZ_coeff", "FullModel_SCZ_SE")],
+res[[3]][dmpList[,1], c("FullModel_SCZ_P", "FullModel_SCZ_coeff", "FullModel_SCZ_SE")])
+colnames(dmpRes)<-c("ProbeID", "DiscoveryCellType", outer(c("P", "Coeff", "SE"), cellTypes, paste, sep = ":"))
+
+
+
+diffLong<-pivot_longer(data.frame(dmpRes[,c(2,4,7,10)]), cols = gsub("\\+|-", "\\.", paste("Coeff", cellTypes, sep = ".")))
+diffLong$name<-gsub("Coeff\\.", "", diffLong$name)
+diffLong$value<-abs(diffLong$value)
+pdf(file.path(resPath, "Plots","ViolinPlotCelltypeEffectsDiscoveryDMPsLMWithinCTs.pdf"), width = 8, height = 4)
+ggplot(diffLong, aes(x = name, y = value, fill = name)) + geom_violin() +
+xlab("Cell Type") + ylab("Mean difference") + 
+stat_summary(fun=mean, geom="point", size=2, color="black") + facet_wrap (~DiscoveryCellType)
+dev.off()
+
+plotLim<-range(dmpRes[,c(4,7,10)])
+
+p1<- subset(dmpRes, DiscoveryCellType == "NeuN+") %>%
+ggplot(aes(x = `Coeff:Double-`, y = `Coeff:NeuN+`)) + geom_point() +
+xlab("Double -ve") + ylab("NeuN +ve") + ggtitle("NeuN+ve DMPs") +
+xlim(plotLim) + ylim(plotLim) +
+geom_hline(yintercept = 0) + 
+geom_vline(xintercept = 0)
+
+p2<- subset(dmpRes, DiscoveryCellType == "NeuN+") %>%
+ggplot(aes(x = `Coeff:Sox10+`, y = `Coeff:NeuN+`)) + geom_point() +
+xlab("Sox10 +ve") + ylab("NeuN +ve") + ggtitle("NeuN+ve DMPs") +
+xlim(plotLim) + ylim(plotLim) +
+geom_hline(yintercept = 0) + 
+geom_vline(xintercept = 0)
+
+p3<- subset(dmpRes, DiscoveryCellType == "Sox10+") %>%
+ggplot(aes(x = `Coeff:Double-`, y = `Coeff:Sox10+`)) + geom_point() +
+xlab("Double -ve") + ylab("Sox10 +ve") + ggtitle("Sox10+ve DMPs") +
+xlim(plotLim) + ylim(plotLim) +
+geom_hline(yintercept = 0) + 
+geom_vline(xintercept = 0)
+
+p4<- subset(dmpRes, DiscoveryCellType == "Sox10+") %>%
+ggplot(aes(x = `Coeff:NeuN+`, y = `Coeff:Sox10+`)) + geom_point() +
+xlab("NeuN +ve") + ylab("Sox10 +ve") + ggtitle("Sox10+ve DMPs") +
+xlim(plotLim) + ylim(plotLim) +
+geom_hline(yintercept = 0) + 
+geom_vline(xintercept = 0)
+
+p5<- subset(dmpRes, DiscoveryCellType == "Double-") %>%
+ggplot(aes(x = `Coeff:NeuN+`, y = `Coeff:Double-`)) + geom_point() +
+xlab("NeuN +ve") + ylab("Double -ve") + ggtitle("Double-ve DMPs") +
+xlim(plotLim) + ylim(plotLim) +
+geom_hline(yintercept = 0) + 
+geom_vline(xintercept = 0)
+
+p6<- subset(dmpRes, DiscoveryCellType == "Double-") %>%
+ggplot(aes(x = `Coeff:Sox10+`, y = `Coeff:Double-`)) + geom_point() +
+xlab("Sox10 +ve") + ylab("Double -ve") + ggtitle("Double-ve DMPs") +
+xlim(plotLim) + ylim(plotLim) +
+geom_hline(yintercept = 0) + 
+geom_vline(xintercept = 0)
```

Comment:
I feel like this warrants some refactoring in the future. These plots are all basically the same, just with different labels and axes. This should be able to be put into a function. `ggarrange` accepts a list of plots, so the function could append plots onto a list (to allow line 240 to still work).